btree.go

```diff
@@ -605,10 +639,43 @@ func (t *BTree) Ascend(iterator ItemIterator) {
 	if t.root == nil {
 		return
 	}
-	t.root.iterate(
-		func(a Item) bool { return true },
-		func(a Item) bool { return true },
-		iterator)
+	t.root.iterate(ascend, nil, nil, false, false, iterator)
+}
+
+// DescendRange calls the iterator for every value in the tree within the range
+// [lessOrEqual, greaterThan), until iterator returns false.
+func (t *BTree) DescendRange(lessOrEqual, greaterThan Item, iterator ItemIterator) {
```

Comment:
I was thinking that the Descend* functions should have the opposite syntax/arguments than the Ascend*.

```go
AscendGreaterOrEqual(pivot)           -> DescendLessOrEqual(pivot)
AscendLessThan(pivot)                 -> DescendGreaterThan(pivot)
Ascend(pivot)                         -> Descend(pivot)
AscendRange(greaterOrEqual, lessThan) -> DescendRange(lessOrEqual, greaterThan) 
```